TASK-86: Misc Cleanup

cristian-aldea opened the pull request:
closes #314

How to test:
- Make sure the following points are addressed from the ticket:
    - 86.1 On Discover/Upcoming/My Events, show “No events found” when queries are empty
    - 86.2 Smaller Mull logo and nav bar on desktop
    - 86.3 Remove all back buttons when on desktop
        - There are exceptions that must be made for this. Namely whenever the onclick is overridden, then I don't hide the button, since it might be used for something other than a browser-type back button
    - 86.4 All buttons and inputs should have feedback when you hover and click
    - 86.5 Remove all share buttons
    - 86.6 All text input limits should match backend varchar limits
    - 86.7 Remove Map button and move the buttons around according to this [mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=3691%3A87700)

cristian-aldea committed:
fix: Show "no event found" on empty home list

ref #314

cristian-aldea committed:
fix(nav): Smaller logo and buttons on desktop

Also, cleaned up styling and added mixins

ref #314

cristian-aldea committed:
fix(back button): Remove back button on desktop

But not removed when onClick is overriden, since that means it will no longer just go back to the previous route, and won't behave like the back button on a browser

ref #314

cristian-aldea committed:
fix(style): Add feedback to hover/click on buttons/forms

ref #314

cristian-aldea committed:
fix(style): Remove share button

ref #314

cristian-aldea committed:
fix(forms): Ensure form limits are the same as db limits

Also, added new restrictions on username and user description

ref #314

cristian-aldea committed:
fix(style): Remove map button and move profile button

ref #314

cristian-aldea committed:
fix(style): Hide scrollbars on most browsers

ref #314

cristian-aldea committed:
fix: Address test issues

ref #314

RGPosadas commented on apps/mull-ui/src/app/components/navigation/top-nav-bar/top-nav-bar.scss:
```diff
@@ -12,38 +13,31 @@
 @media (min-width: $breakpoint-mobile) {
   .top-nav-container {
-    padding: 1em 0;
+    padding: 1rem 0;
```
I personally found that since the Müll logo is smaller, it looked a bit too "squished" towards the left. It looks a bit cleaner with more left-padding, but that's my opinion. Lmk what you think~
```suggestion
    padding: 2rem 0;
```

RGPosadas replied:
Oops I completely put this in the wrong place lol, I meant it here
```scss
.top-nav-logo-container {
    position: absolute;
    left: 2em; // <-- ici
    .top-nav-logo {
      width: 7em;
    }
  }
```

cristian-aldea replied:
No worries I know where to put it. I'll have a look and see how it looks

cristian-aldea replied:
I bumped it up to 1.6rem, I think 2 is a bit much, lemme know what you think when I finish my commit

RGPosadas replied:
Yep that's fine as well

RGPosadas commented on apps/mull-ui/src/app/components/pill-options/pill-options.tsx:
```diff
@@ -17,16 +17,18 @@ export const PillOptions = ({ options, onChange, active }: PillOptionsProps) =>
     <div className="pill-options">
       {options.map((option, idx) => (
-        <div
+        <button
```
Comment for 86.4: I know this is the fix for #309, however it still submits the CreateEvent form once I click on any of the pillbox options.

cristian-aldea replied:
ah you're right, good catch

RGPosadas commented on apps/mull-ui/src/app/components/event-card/event-card.tsx:
```diff
@@ -64,22 +64,22 @@ export const EventCard = ({
         <EventMembers profilePictures={dummyProfilePictures} />
         {/* TODO: Implement share */}
-        <button
+        {/* <button
```
Comment for 86.5: Seems like removing the share button broke the placement/styling of the Join event button on the EventCard when user is not joined
![image](https://user-images.githubusercontent.com/33991121/113464417-7e007b80-93fa-11eb-9c83-e81f02a371df.png)

cristian-aldea replied:
It wasn't the share button, but seems like something else caused this, probably the fact I set the background color of all buttons to be white by default

RGPosadas reviewed (changes requested):
Thank goodness we adopted conventional commits, it really shined here since this ticket touched many different parts. Tests + linting pass, my comment for 86.5 is the only blocker here

- [x] 86.1 On Discover/Upcoming/My Events, show “No events found” when queries are empty
- [x] 86.2 Smaller Mull logo and nav bar on desktop
- [x] 86.3 Remove all back buttons when on desktop
- [x] 86.4 All buttons and inputs should have feedback when you hover and click
- [x] 86.5 Remove all share buttons
- [x] 86.6 All text input limits should match backend varchar limits
- [x] 86.7 Remove Map button and move the buttons around according to this [mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=3691%3A87700)

**Acceptance Criteria**
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data

cristian-aldea committed:
fix: Address pr issues

- Logo further from left
- pill options buttons no longer submit
- white background removed from join button

ref #314

cristian-aldea committed:
fix: Update snapshots

ref #314

RGPosadas reviewed (approved):
Thanks for addressing my comments! Tests + linting pass, LGTM!

- [x] 86.1 On Discover/Upcoming/My Events, show “No events found” when queries are empty
- [x] 86.2 Smaller Mull logo and nav bar on desktop
- [x] 86.3 Remove all back buttons when on desktop
- [x] 86.4 All buttons and inputs should have feedback when you hover and click
- [x] 86.5 Remove all share buttons
- [x] 86.6 All text input limits should match backend varchar limits
- [x] 86.7 Remove Map button and move the buttons around according to this [mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=3691%3A87700)

**Acceptance Criteria**
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data

r-saba commented on apps/mull-ui/src/app/pages/home/discover/discover-page.tsx:
```diff
@@ -22,7 +22,11 @@ export const DiscoverPage = ({ history }) => {
   return (
     <div className="discover-page-tabs-container" data-testid="discover-tab">
-      {eventCards}
+      {eventCards && eventCards.length !== 0 ? (
```
Slightly easier to read, if you agree we could apply this everywhere else
```suggestion
      {eventCards && eventCards.length > 0 ? (
```

r-saba reviewed (approved):
- [X] 86.1 On Discover/Upcoming/My Events, show “No events found” when queries are empty
- [X] 86.2 Smaller Mull logo and nav bar on desktop
- [X] 86.3 Remove all back buttons when on desktop
- [X] 86.4 All buttons and inputs should have feedback when you hover and click
- [X] 86.5 Remove all share buttons
- [X] 86.6 All text input limits should match backend varchar limits
- [X] 86.7 Remove Map button and move the buttons around according to this [mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=3691%3A87700)

**Acceptance Criteria**
- [X] Documentation present where necessary
- [X] Unit and UI tests are implemented for every new component and function
- [X] The application should still work with seeded data

cristian-aldea committed:
fix: Use different conditional for empty event list

ref #314
